### ISSUE TITLE
All tensors not on the same device

### ISSUE THREAD

**Issue by mees** [vimalabs/VIMA#2]
Hi there,
I am trying to run the provided example with following command `python3 scripts/example.py --ckpt=checkpoints/200M.ckpt --device=cuda:0 --partition=placement_generalization --task=visual_manipulation`
I get an error that all tensors are not on the same device.

```
Traceback (most recent call last):
  File "/home/oier/code/VIMA/scripts/example.py", line 506, in <module>
    main(arg)
  File "/home/oier/miniconda3/envs/vima/lib/python3.9/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/home/oier/code/VIMA/scripts/example.py", line 118, in main
    prompt_tokens, prompt_masks = policy.forward_prompt_assembly(
  File "/home/oier/code/VIMA/vima/policy/policy.py", line 163, in forward_prompt_assembly
    batch_word_emb = self.prompt_embedding(word_batch)
  File "/home/oier/miniconda3/envs/vima/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/oier/code/VIMA/vima/nn/prompt_encoder/word_embd.py", line 22, in forward
    x = self._embed_layer(x)
  File "/home/oier/miniconda3/envs/vima/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/oier/miniconda3/envs/vima/lib/python3.9/site-packages/torch/nn/modules/sparse.py", line 158, in forward
    return F.embedding(
  File "/home/oier/miniconda3/envs/vima/lib/python3.9/site-packages/torch/nn/functional.py", line 2199, in embedding
    return torch.embedding(weight, input, padding_idx, scale_grad_by_freq, sparse)
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0! (when checking argument for argument index in method wrapper__index_select)
```

**Comment by mees** [comment]
with `device=cpu` it works though, so probably there are some tensors that are initialized in cpu by default and do not receive the device argument.

**Comment by yunfanjiang** [comment]
Hey @mees, thank you for pointing out this! Will investigate

**Comment by yunfanjiang** [comment]
Hey @mees, fixed per d64df5dcf6fe1bc100fc932914aad53b1b3b4ab4. Feel free to let me know if you encounter further issues. Thanks again!
